### ISSUE TITLE
Can't create files in a parent folder when they're combined in the dock nav tree (ex. tags/[id]/edit)

### ISSUE THREAD

**Issue by hilmarp** [zed-industries/zed#19661]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When the navigation tree looks something like this:

```
|-- package.json
|-- svelte.config.js
|-- src
  |-- routes
    |-- tags
      |-- [id]
        |-- edit
          |-- +page.svelte
          |-- +page.server.ts
```

Where `[id]` is a folder containing only another folder `edit`.

The UI will shorten this to:

```
|-- tags
  |-- [id]/edit
    |-- +page.svelte
    |-- +page.server.ts
```

Then I was trying to create new files in the `[id]` folder and it would always create them in the `edit` folder. And if it was a file with the same name it would just do nothing. When clicking on `[id]/edit` in the UI it would always be targeting `edit`, even when I tried clicking right on to `[id]`.

Easy enough to get around it by creating the files in the terminal, but that might not be for everyone.

Here is a screen recording:

https://github.com/user-attachments/assets/ef6504b5-0e04-41bd-ad90-971b862e3240

### Environment

Zed: v0.158.1 (Zed)
OS: macOS 15.0.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by micahasowata** [comment]
I could not recreate the bug but from the video, there are two possible solutions:

### Using the GUI
1. Right click on the directory
2. Select the "Fold Directory" option
3. If that unfolds directory e.g `edit` from `[id]` then your problem is solved, but I doubt it should
4. Right click again on that same directory, the "Fold Directory" option should now be "Unfold Directory", click on it and your problem should be solved

### Terminal
1. Right click on the directory 
2. Select the open in terminal option
3. Change from the `edit` directory (it very likely that that would be directory that would be opened in the terminal if that option is selected) to the `[id]` directory. A simple `cd ./..` should get that done 
4. Create a simple file in the `[id]` directory, you can use `touch a.txt` to do that, the file explorer would automatically unfold then you can carry on building.

**Comment by micahasowata** [comment]
If you have any more issues or questions, feel free to ask

**Comment by hilmarp** [comment]
Thanks. Yeah it's simple to get around the issue, more just the fact I have to have a workaround. Was using VSCode before which also does this folder collapsing/shortening, but it's no problem creating/dragging files when path is collapsed, it just uncollapses and stops doing the shortening.

**Comment by micahasowata** [comment]
Yeah, I get that. Zed is quite young so workarounds might be only solution to this issue for a while, till the team while the team focuses on larger feature sets and bugs.

But since the workaround has solved your issue, you might want to close the issue.

**Comment by SomeoneToIgnore** [comment]
> But since the workaround has solved your issue, you might want to close the issue.

Erm, no. 
This is a bug and is should be fixed.

**Comment by micahasowata** [comment]
On my Zed, which is the regular Zed for Linux, I don't have this issue. I can fold and unfold directory from the suggestion box. 

I might be wrong, but I think if @hilmarp deletes and reinstalls their version Zed, the issue might be fixed permanently.
